From 34ed32726de8d9923478aafdf67f9cb1e0db4081 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 09 Oct 2022 19:36:18 +0300
Subject: [PATCH] [SW-0] gaudi3: recovery: skip pcie configuration for onix

This patch skips pcie configuration for onix done as part of
recovery sequence.

Change-Id: I27bb6f68fd49e6440cfae2946316d2fa2573197e
---

diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index a4de718..883e357 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -5,10 +5,14 @@
  */
 
 #include <arcboot.h>
+#include <security.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
 void security_rr_close_pcie_config(void) {}
+void security_rr_config_debug_if_lbw(__unused struct rom_rr_ctx *rr_ctx) {}
+void security_rr_config_debug_if_hbw(__unused struct rom_rr_ctx *rr_ctx) {}
+void rom_pcie_recovery(void) {}
 
 uint64_t arcboot_rom_get_timestamp_freq(void)
 {
diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 503de38..7947e14 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -128,7 +128,7 @@
 	regmap_writel(0x1, mmD0_PCIE_WRAP_PCIE_RSP_OK_1);
 }
 
-static void rom_pcie_recovery(void)
+__weak void rom_pcie_recovery(void)
 {
 	uint32_t pcie_ctrl_val = arcboot_rom_fuse_pcie_ctrl_config();
 
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 0ea38eb..0a11fc8 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -777,7 +777,7 @@
 }
 
 #define DEBUG_IF_ENABLE_VAL		0
-void security_rr_config_debug_if_lbw(struct rom_rr_ctx *rr_ctx)
+__weak void security_rr_config_debug_if_lbw(struct rom_rr_ctx *rr_ctx)
 {
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK) == DEBUG_IF_ENABLE_VAL)
@@ -788,7 +788,7 @@
 		security_rr_twi_lbw_config(rr_ctx);
 }
 
-void security_rr_config_debug_if_hbw(struct rom_rr_ctx *rr_ctx)
+__weak void security_rr_config_debug_if_hbw(struct rom_rr_ctx *rr_ctx)
 {
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK) == DEBUG_IF_ENABLE_VAL)
